src/core/etl/src/Flow/ETL/Extractor/GeneratorExtractor.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Extractor;
+
+use Flow\ETL\Extractor;
+use Flow\ETL\FlowContext;
+use Flow\ETL\Rows;
+
+/**
+ * @internal
+ */
+final class GeneratorExtractor implements Extractor
+{
+    /**
+     * @param \Generator<Rows> $rows
+     */
+    public function __construct(private readonly \Generator $rows)
+    {
+    }
+
+    public function extract(FlowContext $context) : \Generator
+    {
+        yield from $this->rows;
```

Comment:
I wonder if we should not actually iterate and just make sure that each result from inner Generator is an instance of Rows

Reply:
I'm not sure if it would not be overkill, cause this class is marked as an internal one (same as `ProcessExtractor`), and it's used only in `MemorySort` which provides a `Rows` class.

Reply:
But since PHP does not have private classes, nothing prevents anyone from using it, and additional iteration should not affect performance more than the previous itereator_to_array solution

Reply:
I still think it's overkill, but added such a check.